probe-rs/src/bin/probe-rs/cmd/dap_server/server/session_data.rs

```diff
@@ -71,7 +75,27 @@ impl SessionData {
 
         let options = config.probe_options().load()?;
         let target_probe = options.attach_probe(lister)?;
-        let target_session = options.attach_session(target_probe, target_selector)?;
+        let target_session = options
+            .attach_session(target_probe, target_selector)
+            .map_err(|operation_error| {
+                match operation_error {
+                    OperationError::AttachingFailed {
+                        source,
+                        connect_under_reset,
+                    } => match source {
+                        probe_rs::Error::Timeout => {
+                            if !connect_under_reset {
+                                DebuggerError::UserMessage(format!("{source} This can happen if the target is not in a state where it can be attached to. A hard reset during attaching might help. If your probe supports this option, try using the `connect_under_reset` option."))
+                            } else {
+                                DebuggerError::UserMessage(format!("{source} This can happen if you are using the `connect_under_reset` option, and your probe does not have support for it."))
```

Comment:
I'm not a grammar guru, especially that english isn't my native language, but to me that comma breaks the sentence in two and it reads like "if you read this error, your probe does not support connect-under-reset regardless of whether you used the option or not".

Reply:
Hmmm ... I hate it when you're right LOL

Reply:
@bugadani Can you please confirm / approve the latest amendment, or perhaps make a specific suggestion on how you think it should be phrased?  I'm definitely not known for my grammar, so appreciate the help :)

Reply:
I think it's a lot more obvious, if maybe a bit verbose :)

Reply:
Yeah, I still think it would make it cleaner if we auto retry without connect_under_reset (see ^^^)